ldap/servers/slapd/back-ldbm/ldbm_search.c

```diff
@@ -1941,10 +1941,18 @@ delete_search_result_set(Slapi_PBlock *pb, back_search_result_set **sr)
     rc = slapi_filter_apply((*sr)->sr_norm_filter, ldbm_search_free_compiled_filter,
                             NULL, &filt_errs);
     if (rc != SLAPI_FILTER_SCAN_NOMORE) {
-        slapi_log_err(SLAPI_LOG_ERR,
-                      "delete_search_result_set", "Could not free the pre-compiled regexes in the search filter - error %d %d\n",
+        slapi_log_err(SLAPI_LOG_ERR, "delete_search_result_set",
+                      "Could not free the pre-compiled regexes in the search filter - error %d %d\n",
+                      rc, filt_errs);
+    }
+
+    rc = slapi_filter_apply((*sr)->sr_norm_filter_intent, ldbm_search_free_compiled_filter, NULL, &filt_errs);
```

Comment:
I'm not very familiar with this part of code, could you please explain a bit how adding `slapi_filter_apply((*sr)->sr_norm_filter_intent...` function call connects with removing the match limit?

Reply:
I think it's an opportunistic fix for a mem leak more likely.

Reply:
Oh, okay! Could be:)

Reply:
Correct, memory leak.  Surprised it got missed with the previous commit